src/util/stop.c

```diff
@@ -49,6 +58,7 @@ int nlopt_stop_f(const nlopt_stopping * s, double f, double oldf)
 int nlopt_stop_x(const nlopt_stopping * s, const double *x, const double *oldx)
 {
     unsigned i;
+    /* TODO: iterative calculation of ‖Δx‖ */
```

Comment:
I guess you'll want something like a `norm_diff(n, x, oldx, w)` function that computes the weighted L1 norm of `x-oldx`.